addons/hr_holidays/models/hr_holidays.py

```diff
@@ -508,21 +507,18 @@ def action_validate(self):
     def action_refuse(self):
         if not self.env.user.has_group('hr_holidays.group_hr_holidays_user'):
             raise UserError(_('Only an HR Officer or Manager can refuse leave requests.'))
-
         manager = self.env['hr.employee'].search([('user_id', '=', self.env.uid)], limit=1)
-        for holiday in self:
-            if holiday.state not in ['confirm', 'validate', 'validate1']:
-                raise UserError(_('Leave request must be confirmed or validated in order to refuse it.'))
-
-            if holiday.state == 'validate1':
-                holiday.write({'state': 'refuse', 'manager_id': manager.id})
-            else:
-                holiday.write({'state': 'refuse', 'manager_id2': manager.id})
-            # Delete the meeting
-            if holiday.meeting_id:
-                holiday.meeting_id.unlink()
-            # If a category that created several holidays, cancel all related
-            holiday.linked_request_ids.action_refuse()
+        wrong_state = self.filtered(lambda r: r.state not in ('confirm', 'validate', 'validate1'))
+        if wrong_state:
+            raise UserError(_('Leave request must be confirmed or validated in order to refuse it.'))
+        validate1 = self.filtered(lambda r: r.state == 'validate1')
+        other = self - validate1
+        validate1.write({'state': 'refuse', 'manager_id': manager.id})
+        other.write({'state': 'refuse', 'manager_id2': manager.id})
+        self.mapped('meeting_id').unlink()
```

Comment:
Just to be coherent with other cascade deletes, we could do meetings = self.mapped(), then if meetings: meetings.unlink .